portfolio/src/main/webapp/script.js

```diff
@@ -22,3 +22,37 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = quote;
 }
+
+async function showFact1(){
+    const responseFromServer = await fetch('/hello');
+    const textFromResponse = await responseFromServer.json();
+
+    const helloContainer = document.getElementById('hello-container1');
+    helloContainer.innerHTML='';
+    helloContainer.appendChild(createListElement(textFromResponse.fact1));
+    //helloContainer.innerText = textFromResponse;
+}
+
+async function showFact2(){
+    const responseFromServer = await fetch('/hello');
+    const textFromResponse = await responseFromServer.json();
+
+    const helloContainer = document.getElementById('hello-container2');
+    helloContainer.innerHTML='';
+    helloContainer.appendChild(createListElement(textFromResponse.fact2));
+}
+
+async function showFact3(){
```

Comment:
There is a lot of repeated code here with minor differences. One way to get around having to repeat code is to create a function that returns another function:

```javascript
function generateAlertCallback(contents) {
  return () => {
    alert(contents);
  };
}

const callback = generateAlertCallback("hello world"); // callback now contains a function.
callback(); // Calling callback() will alert with "hello world".
```

Because these are used as event handlers, the `<button onclick="showFactX()">` can instead look like `<button onclick="generateShowFact(x)">`.